### PR TITLE
Ampq with bunny

### PR DESCRIPTION
**Pull request by gevansBigG**
WIP
This branch carries the integration of bunny into landable for direct ampq messaging.

### CODE REVIEW

**Comment by dkaplan88** [comment]
FYI It looks like the spec because of the dummy initializer to bunny

File: landable.gemspec [hunk 1]
```diff
@@ -49,4 +49,5 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency 'valid_attribute'
   gem.add_development_dependency 'pry'
   gem.add_development_dependency 'faker'
+  gem.add_development_dependency 'bunny'
```
Comment (dkaplan88) [inline landable.gemspec]: I do not think we need Bunny as a dependency as it does not have to be enabled by default. Can we just have bunny be included on the parent app and not in landable itself?

File: lib/generators/templates/landable.rb [hunk 2]
```diff
@@ -51,6 +51,31 @@
 
   # If you want to save a different UserAgent if the request.user_agent is blank, set it here
   # config.blank_user_agent_string = 'blank'
+
```
Comment (dkaplan88) [inline lib/generators/templates/landable.rb]: there are still a couple of references to Hutch in these comments, we can get rid of those now, and update them to Bunny

**Comment by dkaplan88** [comment]
I see you are using a global variable ```MESSAGING_SERVICE``` to initiate Bunny.  

I think we can make this better by initiating Bunny (or any messaging service) via a Landable Initializer.  

Also this way we would not need to require bunny as a dependency to landable and we would not care what messaging service they use. All we should care about is whether they want to use EventPublisher, if they have a message service, and what que they want to use (if any).   

So this would something like:
in confg/initializers/landable.rb in your app
```
config.event_publisher = :enabled # (defaults to false)
config.messaging_service = AmpqMessagingService.new(ampq_configuration) #event_que can be a variable of this messaging service
```

The EventPublisher class looks the same but we can clean it up a bit to be messaging service agnostic.  

Let me know if this makes sense. I also want to hear your thoughts on this approach, and I am totally down to pair on this if you wish :smile:

**Comment by gevansBigG** [comment]
closing - the branch name has a typo

**Pull request closed without merging**